wfe2/wfe.go

```diff
@@ -733,22 +814,30 @@ func (wfe *WebFrontEndImpl) NewAccount(
 		InitialIP:       ipBytes,
 	}
 
+	// TODO(#5545): This can no longer by async once we start treating the new
+	// limiter as the source of truth for rate limits. But for now, this saves
+	// us from eating extra latency for each new account creation.
+	go wfe.checkNewAccountLimits(ctx, ip)
+
 	// Send the registration to the RA via grpc
 	acctPB, err := wfe.ra.NewRegistration(ctx, &reg)
 	if err != nil {
 		if errors.Is(err, berrors.Duplicate) {
 			existingAcct, err := wfe.sa.GetRegistrationByKey(ctx, &sapb.JSONWebKey{Jwk: keyBytes})
 			if err == nil {
 				returnExistingAcct(existingAcct)
+				wfe.refundNewAccountLimits(ctx, ip)
```

Comment:
🤦🏻‍♀️ yep, I'll come up with some kind of solution for this.

Reply:
Alright, I've decided to address this by using `context.WithoutCancel()` on all contexts passed to `ratelimits.Limiter` before passing it onto `ratelimits.source.Set()`/ `Get()`/ `Delete()`. The docs for each `ratelimits.source` method have been updated to included the following:

> Contexts passed to this method should have a timeout or deadline set to prevent the operation from blocking indefinitely.

This should allow rate limit operations to be more predictable since they can only be cancelled for running afoul of a deadline set by the source implementation. Right now we actually rely on the go-redis client itself.